src/natten/flex.py

```diff
@@ -42,6 +42,11 @@
 from .utils import check_all_args
 
 
+def can_run_flex_attention(input_shape) -> bool: 
+    batch_size, *seq_dims, num_heads, head_dim = input_shape
+    seq_length = math.prod(seq_dims)
```

Comment:
WAIT, YOU CAN DO THAT?????

Awesome!!! Can you apply this change to the checks below?

Reply:
@alihassanijr based on this, could we collapse all `flex_na.d(...)` into just one `flex_na(num_seq_modes,...)`? I see this syntactic thing to be the only thing that is hindering us from having a single function that covers all three cases.

Reply:
We --should--, but let's just put that in our list of things to refactor.
It doesn't stop at `flex_na*d` -- we can basically do the same thing for every single `na*d**` API, and not just at the python level.